### PR TITLE
Update All and App package versions

### LINKED ISSUE #8498: Update All and App package versions
**_Roasted Rat Burgers!_** :rat: 🍔 2.1.4 landed a day after we did the 2.1.3 updates. I'll set up a new PR to move `All`/`App` packages to 2.1.4.

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #8498 

I found and fixed seven instances of `App` without a version, too.

### CODE REVIEW

File: aspnetcore/fundamentals/dependency-injection/samples/2.x/DependencyInjectionSample/DependencyInjectionSample.csproj [hunk 1]
```diff
@@ -5,7 +5,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.AspNetCore.App" />
+    <PackageReference Include="Microsoft.AspNetCoreApp" Version="2.1.4" />
```
Comment (scottaddie) [inline aspnetcore/fundamentals/dependency-injection/samples/2.x/DependencyInjectionSample/DependencyInjectionSample.csproj]: Missing the period before "App" in the package name.

File: aspnetcore/fundamentals/repository-pattern/samples/2.x/RepositoryPatternSample/RepositoryPatternSample.csproj [hunk 2]
```diff
@@ -5,7 +5,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.AspNetCore.App" />
+    <PackageReference Include="Microsoft.AspNetCoreApp" Version="2.1.4" />
```
Comment (scottaddie) [inline aspnetcore/fundamentals/repository-pattern/samples/2.x/RepositoryPatternSample/RepositoryPatternSample.csproj]: Missing period in package name

File: aspnetcore/performance/caching/memory/sample/RPcache/RPcache.csproj [hunk 3]
```diff
@@ -5,7 +5,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.AspNetCore.App" />
+    <PackageReference Include="Microsoft.AspNetCoreApp" Version="2.1.4" />
```
Comment (scottaddie) [inline aspnetcore/performance/caching/memory/sample/RPcache/RPcache.csproj]: Missing period in package name

File: aspnetcore/security/authorization/secure-data/samples/starter2.1/ContactManager.csproj [hunk 4]
```diff
@@ -6,7 +6,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.AspNetCore.App" />
+    <PackageReference Include="Microsoft.AspNetCoreApp" Version="2.1.4" />
```
Comment (scottaddie) [inline aspnetcore/security/authorization/secure-data/samples/starter2.1/ContactManager.csproj]: Missing period in package name

File: aspnetcore/security/ip-safelist/samples/2.x/ClientIpAspNetCore/ClientIpAspNetCore.csproj [hunk 5]
```diff
@@ -5,7 +5,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.AspNetCore.App" />
+    <PackageReference Include="Microsoft.AspNetCoreApp" Version="2.1.4" />
```
Comment (scottaddie) [inline aspnetcore/security/ip-safelist/samples/2.x/ClientIpAspNetCore/ClientIpAspNetCore.csproj]: Missing period in package name

File: aspnetcore/signalr/authn-and-authz/sample/SignalRAuthenticationSample.csproj [hunk 6]
```diff
@@ -5,7 +5,7 @@
         <TypeScriptToolsVersion>2.9</TypeScriptToolsVersion>
     </PropertyGroup>
     <ItemGroup>
-        <PackageReference Include="Microsoft.AspNetCore.App" />
+        <PackageReference Include="Microsoft.AspNetCoreApp" Version="2.1.4" />
```
Comment (scottaddie) [inline aspnetcore/signalr/authn-and-authz/sample/SignalRAuthenticationSample.csproj]: Missing period in package name

File: aspnetcore/tutorials/signalr/sample/SignalRChat.csproj [hunk 7]
```diff
@@ -5,7 +5,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.AspNetCore.App" />
+    <PackageReference Include="Microsoft.AspNetCoreApp" Version="2.1.4" />
```
Comment (scottaddie) [inline aspnetcore/tutorials/signalr/sample/SignalRChat.csproj]: Missing period in package name

**Comment by guardrex** [comment]
Lovely ... Find-and-replace ... I need to read a topic on how to use it.

**Comment by scottaddie** [comment]
@guardrex It happens to the best of us.

**Pull request merged**